### ISSUE TITLE
[vuex-setters] Special setters

### ISSUE THREAD

**Issue by louisameline** [vuex-cloud/rfcs#5]
So we have these special values to deal with:
`serverTimestamp`, `increment`, `arrayRemove`, `arrayAdd`

```javascript
// A. current syntax
dispatch('module/extend', { a: { bb: increment(10), cc: 1, dd: 2 }})

// B. proposed syntax (no need to import a function, but need to split things up)
dispatch('module/increment', { 'a.bb': 10 }) // or { a: { bb: 10 }} but this is confusing, it looks like we're doing `a.bb = 10`
dispatch('module/extend', { a: { cc: 1, dd: 2 }})
```

We need to think how vuex-setters will communicate that information to vuex-cloud-firestore's listener (and any other project's listener).

Let's say that vuex-setters exposes an event emitter, in v-c-f it would be like:

```javascript
VuexSetters.on((eventType, payload, helpers) => { })
```
With syntax B, it's pretty straightforward for the listener: the two calls triggers 2 events, with the following `eventType`: `increment` and `extend`.

With syntax A, if we trigger only 1 event of the `extend` type, the listener has to do the work of parsing the payload to see that there is an increment involved (not sure how it would do that exactly). Not great.
To make things simpler for listeners, should action A trigger 2 events of the `increment` and `extend` types?

That's why so far I'm in favor of syntax B.

Note: there this function, I don't really know how it's supposed to be used: https://firebase.google.com/docs/reference/js/firebase.firestore.DocumentReference#isequal

**Comment by mesqueeb** [comment]
I think it's better we discuss event emitters elsewhere. Let's keep this discussion just on **what our syntax will look** like for increment. I'll open issues for the others, until we agree on all of them one by one. 😉 

The ideal case is that people can increment, mutate arrays, update fields to serverTimestamps without having to rely on events. They just dispatch an action and behind the scenes we will update both the local store and remote store appropriately.

**Comment by louisameline** [comment]
Sure Luca, The event emitter is used only for communication between vuex-setters and v-c-f. But we need to keep in mind what will be going on under the hood to make the best decisions at the user API level.

You haven't given me your thoughts on syntax B though?

**Comment by mesqueeb** [comment]
That's right ! I'm outside now and have some freelance stuff to do tonight when I get home ><
I wanna decide on all the issues of the first column of the roadmap before I start thinking of these, because my train of thought will change depending on the issues of the first column.

**Comment by louisameline** [comment]
Alright :)
